hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/ShowSavepointsProcedure.scala

```diff
@@ -54,7 +56,11 @@ class ShowSavepointsProcedure extends BaseProcedure with ProcedureBuilder {
     val commits: util.List[HoodieInstant] = timeline.getReverseOrderedInstants.collect(Collectors.toList[HoodieInstant])
 
     if (commits.isEmpty) Seq.empty[Row] else {
-      commits.toArray.map(instant => instant.asInstanceOf[HoodieInstant].getTimestamp).map(p => Row(p)).toSeq
+      if (limit.isDefined) {
+        commits.stream().limit(limit.get.asInstanceOf[Int]).toArray.map(instant => instant.asInstanceOf[HoodieInstant].getTimestamp).map(p => Row(p)).toSeq
```

Comment:
Can we abstract out the limit function into a common logic in parent class, the parent class is responsible to limit and collect the data frame, and can we add a test for it.

Reply:
ok，I would change it and add ut for it

Reply:
I try to refactor the code for abstracting out limit in parent class，but seems not fit，because would get parameter from sub-class and cannot confirm the order of limit，so I add ut first. at the same time, show procedure commands are not much.

Reply:
try to construnct a 'limit and collect' method in parent class，but had 2 problem to face：

Reply:
I had remove some use less procedures and uts， ptal @danny0405